### PR TITLE
Add span batch logging & metrics

### PR DESCRIPTION
**Pull request by ImTei**
**Description**

As action items for span batch failure modes analysis, this PR adds some new logs and metrics for span batch for
- better visibility to clearly show that the span batch is working well
- easier failure point detection on incidents

**Changes**

- Add more logs and info to specify batch type
- Add min/max inclusion block number for a channel submitted by op-batcher, to figure out when the frames are submitted for a specific channel.
- Add pending safe head ref metric
- Add `DerivedSingularBatches` and `DerivedSpanBatches` metrics: counts for each type of batch derived in op-node.

**Tests**

New metrics are tested in my local devnet.

### CODE REVIEW

File: op-node/rollup/driver/driver.go [hunk 1]
```diff
@@ -27,6 +27,9 @@ type Metrics interface {
 	RecordChannelTimedOut()
 	RecordFrame()
 
+	RecordDerivedSingularBatches()
```
Comment (tynes) [inline op-node/rollup/driver/driver.go]: We we want an individual metric for each batch type? This grows for each new type of batch added. Would it be better to have some sort of single metric that can accept a type?

Comment (trianglesphere) [reply op-node/rollup/driver/driver.go]: I also think that having a batch type label will make it easier to work with the metrics. It's easier to aggregate/slice & dice within a single metric (with labels) than across multiple metrics.

**Review by trianglesphere — COMMENTED** [review]
Logging LGTM. Echoing what tynes said about the metrics.

File: op-node/rollup/driver/driver.go (hunk 1 above)
Comment (protolambda) [reply op-node/rollup/driver/driver.go]: changed it from a `metrics.Event` to a `metrics.EventVec` :+1:

**Comment by protolambda** [comment]
Implemented the minor suggested change to make the batch metric work with a label instead of defining it as two separate metrics.
Also merged in develop, and resolved two minor merge conflict (logging and some stylistic things changed in another span batch PR)

**Review by protolambda — APPROVED** [review]

**Pull request merged**